### ISSUE TITLE
大量仕分けの効率化

### ISSUE THREAD

**Issue by pandeiro245** [pandeiro245/shiwake#1]
* [ ] チェックボックスを用意して一括で親タスクを設定
* [ ] 未分類タスクのフィルタリング

**Comment by gkotaka** [comment]
お世話になります。小鷹と申します。
クラウドワークスでこちらの案件を拝見し、現状の仕様理解を含めて設計を検討しています。
いくつか質問がありますので、お忙しい中恐れ入りますが、ご回答いただければ幸いです。

> チェックボックスを用意して一括で親タスクを設定

こちらですが、「チェックボックスで選択した親タスクに、まとめて同一のカテゴリを設定する」という解釈であっていますでしょうか？

カテゴリ自体のModelはまだ作られていないように見えるのですが、
Modelの設計も含めて行うという形でよいのでしょうか。

また、参考までに、このサービスの具体的な用途を教えていただけると助かります。

"カテゴリ分けした結果を元に、作業時間などを集計・分析したい"
など。

よろしくお願いします。

**Comment by yosimurat** [comment]
初めまして。
クラウドワークスのタスクを始めるにあたって、まずは環境構築を行いたいと思いますが、
自分の開発端末にCloneしてきて、Railsサーバーを起動まではできたのですが、
Sign in したところで、"Client authentication failed due to unknown client, no client authentication included, or unsupported authentication method."でエラーとなってしまいます。

下記以外に、事前準備は他に必要ないでしょうか？

bundle install
rake db:migrate
vim config/settings/development.yml


development.ymlの下記の設定項目あたりも設定すべき項目があれば教えていただければ幸いです。
secret_key_base:
timecrowd:
  client_id: 
  client_secret: 

以上、何卒よろしくお願いします。

**Comment by pandeiro245** [comment]
@yosimurat ご連絡ありがとうございます！
https://timecrowd.net/apidoc
https://timecrowd.net/oauth/applications/new

ここからtoken発行してください！
callbackurlは http://localhost:3000/だとしたら http://localhost:3000/auth/timecrowd/callback
となります。

**Comment by pandeiro245** [comment]
@gkotaka 返信が遅れて失礼しました。

> 「チェックボックスで選択した親タスクに、まとめて同一のカテゴリを設定する」という

はい、その通りです！

> カテゴリ自体のModelはまだ作られていないように見えるのですが、
> Modelの設計も含めて行うという形でよいのでしょうか。

TimeCrowdのタスクは親タスクを設定できるのですがその「親タスク」のことをどこかで「カテゴリ」と表現したかもしれません。つまり、DBを作る必要はなくTimeCrowd APIのみで対応お願いします

**Comment by yosimurat** [comment]
開発環境でSign Inできました。ありがとうございます。

**Comment by pandeiro245** [comment]
@yosimurat 速い！よろしくお願い致します〜！

**Comment by pandeiro245** [comment]
@gkotaka @yosimurat WIPでもPRして頂けましたらレビューするように致しますね

**Comment by gkotaka** [comment]
ご回答ありがとうございます。

> TimeCrowdのタスクは親タスクを設定できるのですがその「親タスク」のことをどこかで「カテゴリ」と表現したかもしれません。
> つまり、DBを作る必要はなくTimeCrowd APIのみで対応お願いします

なるほど。承知いたしました。
「TimeCrowdの親タスク」を一括で設定する機能ということですね。

重ねての質問で恐縮ですが、２点ほど。。

1)  #2 のコミットの内容ですと、最上位タスクにだけチェックボックスが付くようですが、
　「右側のプルダウンメニューで選択しているタスクの親タスクを変更する」という認識であっていますか？

たとえば、「変更後の親タスクB」というタスクに変更する場合、

　[レ] 親タスクA 「 親タスクA - 子タスクA 」

という状態の場合には、「子タスクA」の親タスクが「変更後の親タスクB」に変更になる。

　[レ] 親タスクA 「  」(未選択)

という状態の場合には、「親タスクA」の親タスクが「変更後の親タスクB」に変更になる。

※[レ]はチェックボックス、「　　」はプルダウンメニューを表します。

2) 「親タスクを新規作成して割当」については、
　既存のタスクに設定する親タスクを、既存タスクの中から選べるだけでなく、新規に作成することもできるようにするということでしょうか？
　(「チェックボックスを用意して一括で親タスクを設定」の拡張機能)

　それとも、既存のタスクは関係なく、純粋に「新しいタスクを作れるようにする機能」のことでしょうか？

よろしくお願いします。

**Comment by yosimurat** [comment]
> 未分類タスクのフィルタリング
こちらは、検索窓のようなものを画面に設けて、タイトルで未分類タスクから絞り込める様な仕様でよろしいでしょうか？

**Comment by pandeiro245** [comment]
@gkotaka 

>「右側のプルダウンメニューで選択しているタスクの親タスクを変更する」という認識であっていますか？

![image](https://cloud.githubusercontent.com/assets/125965/13836907/9c6c44e0-ec49-11e5-8958-b3055fc3d1ed.png)

はい。上図でいうところでいうと「11837: 245cloud（ニシコクラウド）」の右側の黒いボタンを押すと
下４つのチェックボックスがONになっているタスクの親タスクがそれに設定されて一覧から消えるイメージです。

> 既存のタスクに設定する親タスクを、既存タスクの中から選べるだけでなく、新規に作成することもできるようにするということでしょうか？

そうです。「親タスクを新規で作成」というリンクを設置し、クリックするとモーダルが出て、タスク名入力と最上位タスクとして登録するチェックボックスがあり、保存を押すとそれが作成され、それが親タスクとして登録されればOKです

**Comment by pandeiro245** [comment]
@yosimurat はい、タイトルとURLで検索したいです。（難しければタイトルのみでOK）

**Comment by gkotaka** [comment]
> pandeiro245 さま
画像付きでご丁寧にありがとうございます。
イメージがわきました！

**Comment by yosimurat** [comment]
未分類タスクのフィルタリング対応して、PR送りました。
よろしくお願いします。

**Comment by pandeiro245** [comment]
@yosimurat マージしてこのissueの「未分類タスクのフィルタリング（キーワード検索）」をdoneにし、クラウドワークス承認を行ないました！引き続きこれぐらいの粒度でPRお願いします！！

**Comment by yosimurat** [comment]
承認ありがとうございます！

> フィルタリングロジックの保存
> 時間もあまりないですが、kintoneタスクの方の環境構築のメドがたたないため、こちらの仕様のタスクをすすめて行きたいと思います。

具体的な仕様の確認ですが、「検索条件を保存する」等のボタンを設けてフィルタリングした検索条件、つまりキーワードをDB上に記録しておいて、後で呼び出せるようにするとの認識でよろしいでしょうか？
検索キーワードはユーザーごとに複数件、登録できるものとします。
登録したものは「保存した検索条件」などのボタンなどから呼び出せるものとします。

以上で問題なければ実装を進めます。
よろしくお願いします。

**Comment by pandeiro245** [comment]
@yosimurat その仕様で問題ないです。よろしくお願いたします！

**Comment by yosimurat** [comment]
#12 を対応するには、#8 「チェックボックスを用意して一括で親タスクを設定」も一緒に対応するひつようがありますね。
#8 もこちらで引き取って、４月からとりかかりますが、よろしいでしょうか？

**Comment by pandeiro245** [comment]
@yosimurat そうですね。https://github.com/mashupwork/kintone_portal/issues/30 を優先しつつ、お願いできると幸いです。
